user-operator/src/main/java/io/strimzi/operator/user/model/KafkaUserModel.java

```diff
@@ -387,11 +387,15 @@ protected byte[] decodeFromSecret(Secret secret, String key) {
      * @return The secret.
      */
     protected Secret createSecret(Map<String, String> data) {
+        Map<String, String> labels = this.labels.toMap();
+
+        labels = Labels.filterLabels(Util.mergeLabelsOrAnnotations(labels, templateSecretLabels)).toMap();
+
         return new SecretBuilder()
                 .withNewMetadata()
                     .withName(getSecretName())
                     .withNamespace(namespace)
-                    .withLabels(Util.mergeLabelsOrAnnotations(labels.toMap(), templateSecretLabels))
+                    .withLabels(Util.mergeLabelsOrAnnotations(labels, templateSecretLabels))
```

Comment:
Why is this needed? The labels are taked through the `fromResource` method that already filtered them. That is how CO uses it. Why is it not good enough here?

Reply:
In other words, it already applies the exclusion, doesn't it?

Reply:
Yeah but in the constructor of `KafkaUserModel` we do 
```java
protected KafkaUserModel(String namespace, String name, Labels labels, String secretPrefix) {
        this.namespace = namespace;
        this.name = name;
        this.labels = labels.withKubernetesName(KAFKA_USER_OPERATOR_NAME)
            .withKubernetesInstance(name)
            .withKubernetesPartOf(name)
            .withKubernetesManagedBy(KAFKA_USER_OPERATOR_NAME);
        this.secretPrefix = secretPrefix;
    }
```
So the thing is that we filter labels in the `fromCrd` method scope
```java
public static KafkaUserModel fromCrd(KafkaUser kafkaUser,
                                         String secretPrefix,
                                         boolean aclsAdminApiSupported) {
        KafkaUserModel result = new KafkaUserModel(kafkaUser.getMetadata().getNamespace(),
                kafkaUser.getMetadata().getName(),
                Labels.fromResource(kafkaUser).withStrimziKind(kafkaUser.getKind()),
                secretPrefix);
               ...
               ...
```
but after that in the constructor, we add labels to the model, which seems weird to me. 
---
anyway as you pointed on that line there should be only `labels` which is filtered in L:392.

Reply:
We get the labels from the custom resource, we filter them based on the exclusions ... and then we add our own labels. That is expected. You can never filter all labels. Sure, if you dislike one of the labels we set, you might not be able to change it. But that is life and we might need to live with it. But as I said for the previous variant, I think we should aim for consistent solution everywhere and not doing something else in every part of the project.

Reply:
Well in that case I think filtering in the createSecret is not needed. And labels, which we add i.e., `app.kubernetes.io/...` will be there...and we need to live with it :D. Thus we would have a consistent solution for all projects.